src/highlight_processor.c

```diff
@@ -0,0 +1,200 @@
+#include "result_processor.h"
+#include "fragmenter.h"
+#include "value.h"
+#include "util/minmax.h"
+#include <ctype.h>
+
+/**
+ * Attempts to fragmentize a single field from its offset entries. This takes
+ * the field name, gets the matching field ID, retrieves the offset iterator
+ * for the field ID, and fragments the text based on the offsets. The fragmenter
+ * itself is in fragmenter.{c,h}
+ *
+ * Returns true if the fragmentation succeeded, false otherwise.
+ */
+static int fragmentizeOffsets(const RSSearchRequest *req, const char *fieldName,
```

Comment:
Do not pass the whole request if you only need the spec from it. Also, the spec object might change on concurrent switches, you need to get the query execution context or the processor context here